playbooks/robusta_playbooks/k8s_resource_enrichments.py

```diff
@@ -43,32 +46,103 @@ def to_pod_row(pod: Pod, cluster_name: str) -> List:
         pod.status.phase,
     ]
 
-
-@action
-def related_pods(event: KubernetesResourceEvent):
-    """
-    Return the list of pods related to that k8s resource.
-    For example, return all pods of a given k8s Deployment
-
-    Supports Deployments, ReplicaSets, DaemonSets, StatefulSets and Pods
-    """
-    resource = event.get_resource()
-    if resource.kind not in supported_resources:
+def get_related_pods(resource) -> list[Pod]:
+    kind: str = resource.kind or ""
+    if kind not in supported_resources:
         raise ActionException(
-            ErrorCodes.RESOURCE_NOT_SUPPORTED, f"Related pods is not supported for resource {resource.kind}"
+            ErrorCodes.RESOURCE_NOT_SUPPORTED, f"Related pods is not supported for resource {kind}"
         )
 
-    if resource.kind == "Job":
+    if kind == "Job":
         job_pods = get_job_all_pods(resource)
         pods = job_pods if job_pods else []
-    elif resource.kind == "Pod":
+    elif kind == "Pod":
         pods = [resource]
-    elif resource.kind == "Node":
+    elif kind == "Node":
         pods = Pod.listPodForAllNamespaces(field_selector=f"spec.nodeName={resource.metadata.name}").obj.items
     else:
         selector = build_selector_query(resource.spec.selector)
         pods = PodList.listNamespacedPod(namespace=resource.metadata.namespace, label_selector=selector).obj.items
 
+    return pods
+
+def to_pod_obj(pod: Pod, cluster: str) -> Dict[str,Any]:
+    resource_requests = pod_requests(pod)
```

Comment:
if we are already passing 70% of the pods yaml why don't we just pass the whole yaml of the pod and parse it on the the frontend?
it's less code in runner and more versatile for frontend?